poutyne/framework/callbacks/wandb_logger.py

```diff
@@ -0,0 +1,256 @@
+        # pylint: disable-msg=too-many-locals
+
+        super().__init__(batch_granularity=batch_granularity)
+
+        if wandb is None:
+            raise ImportError("WandB needs to be installed to use this callback.")
+
+        anonymous_lut = {True: "allow", False: None}
+        self._wandb_init = dict(
+            name=name,
+            group=group,
+            config=config,
+            project=project,
+            id=version or run_id,
+            dir=save_dir,
+            resume="allow",
+            anonymous=anonymous_lut.get(anonymous, anonymous),
+        )
+
+        if experiment is None:
+
+            if offline:
+                os.environ["WANDB_MODE"] = "dryrun"
+
+            if wandb.run is None:
+                self.run = wandb.init(**self._wandb_init)
+            else:
+                warnings.warn(
+                    "There is already a wandb run experience running. This callback will reuse this run. If you want "
+                    "to start a new one stop this process and call `wandb.finish()` before starting again."
+                )
+                self.run = wandb.run
+        else:
+            self.run = experiment
+
+        self.run.config.update({"run_id": self.run.id})
+        self.log_gradient_frequency = log_gradient_frequency
+        self.training_batch_shape = training_batch_shape
+        self.checkpoints_path = checkpoints_path
+        if initial_artifacts_paths is not None:
+            self._log_artifacts(initial_artifacts_paths, name="Initial-artifacts", artifact_type="Miscellaneous")
+
+    def _watch_gradient(self) -> None:
+        """
+        Activate wandb gradient watching.
+        """
+        self.run.watch(self.model.network, log="all", log_freq=self.log_gradient_frequency)
+
+    def _save_architecture(self) -> None:
+        """
+        Save architecture.
+        """
+        dummies_batch = torch.randn(self.training_batch_shape)
+        save_path = self.run.dir + "/" + self.run.name + "_model.onnx"
+        torch.onnx.export(self.model.network, dummies_batch, save_path)
+        self.run.save(save_path)
+
+    def on_train_begin(self, logs: Dict):
+        super().on_train_begin(logs)
+        if self.log_gradient_frequency is not None:
+            self._watch_gradient()
+
+        if self.training_batch_shape is not None:
+            self._save_architecture()
+
+    def log_config_params(self, config_params: Dict) -> None:
+        """
+        Args:
+            config_params Dict:
+                Dictionary of config parameters of the training to log, such as number of epoch, loss function,
+                optimizer etc.
+        """
+        self.run.config.update(config_params)
+
+    def _on_train_batch_end_write(self, batch_number: int, logs: Dict) -> None:
+        """
+        Log the batch metric.
+        """
+        if self.batch_granularity:
+            train_metrics = {key: value for (key, value) in logs.items() if "val_" not in key}
```

Comment:
Why do you remove validation metrics?

Reply:
I only save validation metrics at the end of epochs.

Reply:
I changed it to save both.

Reply:
Actually, there is not supposed to have validation mewtrics in on_train_batch_end... So all the code here related to that is useless.